src/test/java/uk/gov/hmcts/reform/wacaseeventhandler/services/jobservices/SetStateToProcessedMessageJobTest.java

```diff
@@ -0,0 +1,82 @@
+import static org.mockito.Mockito.when;
+
+@ExtendWith({MockitoExtension.class})
+
+
+public class SetStateToProcessedMessageJobTest {
+    @Mock
+    private CaseEventMessageRepository caseEventMessageRepository;
+
+    private SetStateToProcessedMessageJob setStateToProcessedMessageJob;
+
+    private final List<String> messageIds = List.of("messageId_1", "messageId_2", "messageId_3");
+
+    @BeforeEach
+    void setUp() {
+        setStateToProcessedMessageJob = new SetStateToProcessedMessageJob(
+            caseEventMessageRepository,
+            messageIds
+        );
+    }
+
+    @Test
+    void should_be_able_to_run_message_state_message_job() {
+        assertFalse(setStateToProcessedMessageJob.canRun(JobName.FIND_PROBLEM_MESSAGES));
+        assertFalse(setStateToProcessedMessageJob.canRun(JobName.RESET_PROBLEM_MESSAGES));
+        assertFalse(setStateToProcessedMessageJob.canRun(JobName.RESET_NULL_EVENT_TIMESTAMP_MESSAGES));
+        assertTrue(setStateToProcessedMessageJob.canRun(JobName.SET_STATE_TO_PROCESSED_ON_MESSAGES));
+    }
+
+    @Test
+    void should_return_empty_response_for_empty_message_ids() {
+        assertTrue(new SetStateToProcessedMessageJob(caseEventMessageRepository, null)
+                       .run().isEmpty());
+        assertTrue(new SetStateToProcessedMessageJob(caseEventMessageRepository, List.of())
+                       .run().isEmpty());
+    }
+
+    @Test
+    void should_return_empty_response_for_empty_unprocessable_messages() {
+        when(caseEventMessageRepository.findByMessageId(messageIds)).thenReturn(List.of());
+        assertTrue(setStateToProcessedMessageJob.run().isEmpty());
+
+        when(caseEventMessageRepository.findByMessageId(messageIds))
+            .thenReturn(List.of(new CaseEventMessageEntity().buildMessage("messageId_1", MessageState.NEW),
+                                new CaseEventMessageEntity().buildMessage("messageId_2", MessageState.READY),
+                                new CaseEventMessageEntity().buildMessage("messageId_3", MessageState.PROCESSED)));
+        assertTrue(setStateToProcessedMessageJob.run().isEmpty());
+    }
+
+    @Test
+    void should_return_message_id_list_response_for_setting_message_state() {
+
+        CaseEventMessageEntity unprocessableMessage = new CaseEventMessageEntity()
+                                                          .buildMessage("messageId_3", MessageState.UNPROCESSABLE);
+
+        when(caseEventMessageRepository.findByMessageId(messageIds))
+            .thenReturn(List.of(new CaseEventMessageEntity().buildMessage("messageId_1", MessageState.NEW),
+                                new CaseEventMessageEntity().buildMessage("messageId_2", MessageState.READY),
+                                unprocessableMessage));
+
+        assertEquals(messageIds, setStateToProcessedMessageJob.run());
+        assertEquals(MessageState.PROCESSED,unprocessableMessage.getState());
```

Comment:
surely this should be checking the list of messages returned from the `run()` method?